cmake/FileUtil.cmake

```diff
@@ -141,6 +141,18 @@ function(ly_file_read path content)
     set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${path})
 endfunction()
 
+#! ly_file_read: wrap ly_file_read but store the file in a cache to avoid 
+#  extra reads.
+function(ly_file_read_cache path content)
+    unset(file_content)
+    set(file_cache_var_name "O3DE_FILE_CACHE_${path}")
+    get_property(file_content GLOBAL PROPERTY ${file_cache_var_name})
+    if(NOT file_content)
+        ly_file_read(${path} file_content)
+        set_property(GLOBAL PROPERTY ${file_cache_var_name} ${file_content})
+    endif()
+    set(${content} ${file_content} PARENT_SCOPE)
+endfunction()
```

Comment:
I suggest the path is normalized in case it is slightly different(case differences on Windows or backslash/forward slash issues)